src/app/youtube/services/search.service.ts

```diff
@@ -1,40 +1,65 @@
 import { Injectable } from '@angular/core'
-import { HttpClient } from '@angular/common/http'
 import { Subject } from 'rxjs'
-import { IVideosResponse } from '../models/videos-response.model'
-import { IVideoItem } from '../models/video-item.model'
+import {
+  debounceTime,
+  distinctUntilChanged,
+  map,
+  pluck,
+  switchMap,
+  tap
+} from 'rxjs/operators'
+import { IVideoStatsItem } from '../models/video-item.model'
+import { YoutubeHttpService } from './youtube-http.service'
 
 @Injectable({
-  providedIn: 'root',
+  providedIn: 'root'
 })
 export class SearchService {
-  private url = './assets/response.json'
+  private videosQuery$$ = new Subject<string>()
 
-  response?: IVideosResponse
+  videoItems: IVideoStatsItem[] = []
 
-  videos: IVideoItem[] = []
+  private videos$$ = new Subject<IVideoStatsItem[]>()
 
-  videos$ = new Subject<IVideoItem[]>()
+  private videoDetails$$ = new Subject<IVideoStatsItem | null>()
 
-  videoDetails$ = new Subject<IVideoItem | undefined>()
+  videos$ = this.videos$$.asObservable()
 
-  constructor(private http: HttpClient) {
-    this.getVideos()
+  videoDetails$ = this.videoDetails$$.asObservable()
+
+  constructor(private httpService: YoutubeHttpService) {
+    this.videosQuery$$
+      .pipe(
+        debounceTime(1000),
+        distinctUntilChanged(),
+        switchMap(query => {
+          return this.httpService.getVideos(query)
+        }),
+        pluck('items'),
+        map(items => items.map(v => v.id.videoId).join(',')),
+        switchMap(ids => {
+          return this.httpService.getStatistics(ids)
+        }),
```

Comment:
take a look at this rule https://eslint.org/docs/rules/arrow-body-style. Be consistent in your codestyle - either use {} with return in arrow function or not, don't mix it.

Reply:
add rule